File: argentum-django/api/tests/data/product_ranges.py
from api.models.product_range import ProductRange
from api.tests.data.products import TestProducts
from api.tests.utils.test_objects import TestObjects


class TestProductRanges(TestObjects):
    MODEL = ProductRange

    JUST_WATER = MODEL(
        id=1,
        name='Just water'
    )

    EVERYTHING = MODEL(
        id=2,
        name='Everything'
    )

    ALL = [JUST_WATER, EVERYTHING]

    # Models below are not stored in the DB, but rather used for POST deserialization testing.

    JUST_COKE = MODEL(
        id=3,
        name='Just coke'
    )

    @classmethod
    def init(cls):
        super().init()
        for product_range in cls.ALL:
            product_range.create_permission()

    @classmethod
    def post_init(cls):
        cls.JUST_WATER.products.set([TestProducts.WATER])
        cls.EVERYTHING.products.set([TestProducts.WATER, TestProducts.COKE])

File: argentum-ui/src/app/common/rest-service/test-data/product-ranges.ts
import {ProductRange} from '../../model/product-range';
import {Products} from './products';

export namespace ProductRanges {
  export const JUST_WATER_META = new ProductRange(13010, 'Just water', undefined);
  export const JUST_WATER = new ProductRange(13010, 'Just water', [Products.WATER]);
  export const EVERYTHING_META = new ProductRange(13020, 'Everything', undefined);
  export const EVERYTHING = new ProductRange(13020, 'Everything', [Products.WATER, Products.COKE]);

  export const ALL_META = [JUST_WATER_META, EVERYTHING_META];
  export const ALL = [JUST_WATER, EVERYTHING];

  export const JUST_COKE_META = new ProductRange(13030, 'Just coke', undefined);
  export const JUST_WATER_PATCHED_META = new ProductRange(13011, 'Just aqua', undefined);
}
